db/mongoose.js

```diff
@@ -1,13 +1,35 @@
 const mongoose = require("mongoose");
 const keys = require("../config/keys");
-mongoose.connect(
-  keys.mongoURI,
-  { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false, useCreateIndex: true },
-  function (err) {
-    if (err) {
-      console.log(err);
-    } else {
-      console.log("MongoDB Connected");
-    }
-  }
-);
+mongoose
+  .connect(keys.mongoURI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    dbName: "Messportal",
+    useCreateIndex: true,
+    useFindAndModify: false
+  })
+  .then(() => {
+    console.log("mongodb connected");
+  })
+  .catch((e) => {
+    console.log(e.message);
+    console.log("error in connecting to monogdb");
+  });
+
+mongoose.connection.on("connected", () => {
+  console.log("mongoose connected to db");
+});
+
+mongoose.connection.on("error", (err) => {
+  console.log(err.message);
+  console.log("Error in connecting mongoose to monogdb");
+});
+
+mongoose.connection.on("disconnected", () => {
+  console.log("Mongoose Disconnected");
+});
+
+process.on("SIGINT", async () => {
+  await mongoose.connection.close();
+  process.exit(0);
+});
```

Comment:
Could you brief a little more about this?

Reply:
when the server is closed forcefully or because of some error, the transactions that are mode using mongoose are also closed instantaneously which causes the loss of data. so it just takes care of the mongoose process to finish and then close the program.